pkg/controller/machine.go

```diff
@@ -271,7 +271,7 @@ func (c *controller) updateMachineState(machine *v1alpha1.Machine, node *v1.Node
 					c.updateMachineStatus(machine, lastOperation, currentStatus)
 
 				} else*/
-			if machine.Status.LastOperation.Description == "Creating machine on cloud provider" && len(machine.Status.Conditions) > 1 && machine.Status.Conditions[len(machine.Status.Conditions)-1].Status == "True" {
+			if len(machine.Status.Conditions) > 1 && machine.Status.Conditions[len(machine.Status.Conditions)-1].Status == "True" {
```

Comment:
Can we change the fetching of the machine status condition by the condition type (instead of index based), to avoid mis match if the conditions go haywire?